blocks/chat-cta/chat-cta.js

```diff
@@ -1,5 +1,27 @@
-const init = async (el) => {
-  const cta = el.children[0].children[0];
+export default async function init(el) {
+  const block = el;
+
+  if (el.dataset?.content) {
+    const contentUrl = el.dataset.content;
+    const resp = await fetch(contentUrl);
+    if (!resp.ok) return;
+    const text = await resp.text();
+    if (text) {
+      const parser = new DOMParser();
+      const doc = parser.parseFromString(text, 'text/html');
+      const ctaBody = doc.querySelector('.chat-cta > div');
+      if (!ctaBody) return;
+      block.append(ctaBody);
+      const link = ctaBody.querySelector('a');
+      const url = new URL(link.href);
+      link.dataset.modalPath = url.pathname;
+      link.dataset.modalHash = url.hash;
+      link.href = url.hash;
```

Comment:
Super nit: destructuring
```suggestion
      const { pathname, hash } = new URL(link.href);
      link.dataset.modalPath = pathname;
      link.dataset.modalHash = hash;
      link.href = hash;
```